### PR TITLE
Add unfollow frontend

### PR DESCRIPTION
**Pull request by ArianaGu**
Since the http delete request does not support a body in flutter (it works when I tested with Postman however), so I modify the backend to provide a view to allow using post method to unfollow.
Now the people page shows all the users, for the users that have already been followed by the current user, the button would be `unfollow`, otherwise it would be `follow`. 
image
If the user clicked `follow` , the page would refresh itself and the button would change to `unfollow`.

### CODE REVIEW

File: flutter/Graminsta/lib/main.dart [hunk 1]
```diff
@@ -30,15 +32,12 @@ class _HomePageState extends State<HomePage> {
...
+  static List<Widget> _widgetOptions = <Widget>[
     Text(
       'Index 0: Home',
       style: optionStyle,
     ), //change this into the timeline page.
-    Text(
-      'Index 1: Person',
-      style: optionStyle,
-    ), //change this into the gallery page.
+    UsersList(),
```
Comment (ArianaGu) [inline flutter/Graminsta/lib/main.dart]: This page was intended to be gallery, I changed it to show people because I think personal gallery page can be jumped to by clicking the user listed on this users list page.

File: flutter/Graminsta/lib/models/user.dart [hunk 2]
```diff
@@ -4,16 +4,20 @@ class User {
   final String username;
   final String firstName;
   final String lastName;
+  bool isFollowing;
 
-  User({this.id, this.username, this.firstName, this.lastName});
+  User({this.id, this.username, this.firstName,
+    this.lastName, this.isFollowing});
```
Comment (Time1ess) [inline flutter/Graminsta/lib/models/user.dart]: Add comma before `}`

File: flutter/Graminsta/lib/post/follow_service.dart [hunk 3]
```diff
@@ -13,10 +12,32 @@ class FollowService {
 
   static Future<bool> follow(int targetUserID) async {
     final response = await http.post("post/follow/",
-        body: json.encode({
-          "target_user": targetUserID,
-        }));
-    if (response.statusCode == 209) {
+        body: {
+          "target_user": targetUserID.toString(),
+        });
```
Comment (Time1ess) [inline flutter/Graminsta/lib/post/follow_service.dart]: comma after `}`

File: backend/graminsta/post/urls.py [hunk 4]
```diff
@@ -4,11 +4,14 @@
...
+from .views import (PostRecordView, FollowView,
+                    UnfollowView, FollowingView)
 
 
 app_name = 'post'
 urlpatterns = [
     path('', PostRecordView.as_view(), name='post'),
     path('follow/', FollowView.as_view(), name='follow'),
-    path('follow/<user_id>/', FollowView.as_view(), name='following_people')]
+    path('unfollow/', UnfollowView.as_view(), name='unfollow'),
```
Comment (Time1ess) [inline backend/graminsta/post/urls.py:15]: Just a FYI, usually in RESTful API, to remove a following relationship, you can 

```
DELETE /api/follow/5
```

Which means deleting the relationship with id 5

But I think this is OK to compromise  given our deadline.

**Review by Time1ess — APPROVED** [review]

File: backend/graminsta/post/urls.py (hunk 4 above)
Comment (ArianaGu) [reply backend/graminsta/post/urls.py:15]: Got it. But does this mean that I will need another view to post the target_user_id and fetch the id of the relationship? Only after fetch the id can I use the delete method, since I do not have the relationship id in frontend, so I chose this simple way to delete for now. I would modify it if I have time later.

Comment (Time1ess) [reply backend/graminsta/post/urls.py:15]: In my mind, in your followers page, the data you received from backend should be something like:
```
[
{
  id: 1,
  target_user: {
    id: 111,
    first_name: xx
    last_name: yy
  }
},
{
  id: 2,
  target_user: {
    id: 222,
    first_name: xx
    last_name: yy
  }
}
]
```
 Then you should have enough information both for rendering UI and completing the unfollow action.

File: backend/graminsta/post/urls.py [hunk 5]
```diff
@@ -4,14 +4,14 @@
...
 
 app_name = 'post'
 urlpatterns = [
     path('', PostRecordView.as_view(), name='post'),
     path('follow/', FollowView.as_view(), name='follow'),
     path('unfollow/', UnfollowView.as_view(), name='unfollow'),
     path('following/<user_id>/', FollowingView.as_view(),
-         name='following_people')]
+         name='following_people'),
+    path('timeline/', TimelineView.as_view(), name='timeline')]
```
Comment (Time1ess) [inline backend/graminsta/post/urls.py:18]: Why is this line here? Maybe you added it accidentally?

**Review by Time1ess — APPROVED** [review]

File: backend/graminsta/post/urls.py (hunk 4 above)
Comment (ArianaGu) [reply backend/graminsta/post/urls.py:15]: Got it, I think I should modify the serializer to do so.

**Review by Time1ess — APPROVED** [review]

**Review by Time1ess — APPROVED** [review]

**Review by shiyi001 — APPROVED** [review]
